File: Java_Learning/Library Management System/LMS/src/Library.java
import java.util.ArrayList;

class Library {
     static int bookCount = 0; // Static counter for books
    
     ArrayList<Book> books = new ArrayList<>(); // List of books
    
     ArrayList<User> users = new ArrayList<>(); // List of users

    // Add a new book to the library
    public void addBook(String title, String author) {
        Book newBook = new Book(++bookCount, title, author);
        books.add(newBook);
        System.out.println("Book added: " + title);
    }

    // Add a new user to the library
    public void addUser(String name) {
        User newUser = new User(users.size() + 1, name);
        users.add(newUser);
        System.out.println("User added: " + name);
    }

    // Display all books
    public void displayBooks() {
        System.out.println("\nLibrary Books:");
        for (Book book : books) {
            System.out.println("ID: " + book.getId() + ", Title: " + book.getTitle() + ", Author: " + book.getAuthor() + ", Issued: " + (book.isIssued() ? "Yes" : "No"));
        }
    }

    // Issue a book
    public void issueBook(int bookId) {
        for (Book book : books) {
            if (book.getId() == bookId) {
                book.issueBook();
                return;
            }
        }
        System.out.println("Book ID " + bookId + " not found.");
    }

    // Return a book
    public void returnBook(int bookId) {
        for (Book book : books) {
            if (book.getId() == bookId) {
                book.returnBook();
                return;
            }
        }
        System.out.println("Book ID " + bookId + " not found.");
    }
}

File: Python_Learning/LMS_in_Python/Library.py
from Book import Book
from User import User

class Library():
    
    
    def __init__(self):
        self.bookCount = 0
        self.books = []
        self.users = []
    
    def addBook(self,title,author):
        self.bookCount += 1
        newBook = Book(self.bookCount,title,author)
        self.books.append(newBook)
        print(f"Book added : {title}")
    
    def addUser(self,name):
        newUser = User(len(self.users) , name)
        self.users.append(newUser)
        print(f"User {name} added")
    
    def displaybooks(self):
        print("Library Books : ")
        if not self.books:
            print("No books available")
        else:
            for book in self.books:
                print(f"ID: {book.getId()} | Title: {book.getTitle()} | Author: {book.getAuthor()}")

    def issueBook(self, bookid):
        for book in self.books:
            if book.getId() == bookid:
                book.issueBook()
                return
        print(f"Book id {bookid} not found")
    
    def returnBook(self,bookid):
        for book in self.books:
            if book.getId() == bookid:
                book.returnBook()
                return
        print(f"Bookid {bookid} is not found")
        